app/controllers/session_controller.rb

```diff
@@ -332,6 +333,34 @@ def create
     end
   end
 
+  def passkey_challenge
+    render json: DiscourseWebauthn.stage_challenge(current_user, secure_session)
+  end
+
+  def passkey_auth_perform
+    raise Discourse::NotFound unless SiteSetting.experimental_passkeys
+
+    params.require(:publicKeyCredential)
```

Comment:
Hmm is there a reason the param is camel case instead of snake case?

Reply:
Yes, using camel case here to match the webauthn specs. This credential is created by the browser (via `navigator.credentials`), we could transform it to snake case if we wanted, but it feels more appropriate to stick to the naming established by the spec, see https://w3c.github.io/webauthn/#sctn-verifying-assertion for more context.